packages/flutter_tools/lib/src/migrations/swift_package_manager_integration_migration.dart

```diff
@@ -1023,24 +981,11 @@ class ParsedNativeTarget {
 /// Representation of data parsed from PBXProject section in Xcode project's
 /// project.pbxproj.
 class ParsedProject {
-  ParsedProject._(
-    this.data,
-    this.identifier,
-    this.packageReferences,
-  );
-
-  factory ParsedProject.fromJson(String key, Map<String, Object?> data) {
-    final List<String> parsedChildren = <String>[];
-    if (data['packageReferences'] is List<Object?>) {
-      for (final Object? item in data['packageReferences']! as List<Object?>) {
-        if (item is String) {
-          parsedChildren.add(item);
-        }
-      }
-      return ParsedProject._(data, key, parsedChildren);
-    }
-    return ParsedProject._(data, key, null);
-  }
+  ParsedProject.fromJson(this.identifier, this.data)
+      : packageReferences = switch (data['packageReferences']) {
+          final List<Object?> references => references.whereType<String>().toList(),
+          _ => null,
+        };
```

Comment:
I like this change 👍. When I encounter the `switch`, I already have sufficient context for its purpose, making it easier to parse. This change also saves a boiler plate constructor and a `factory` constructor.